pkg/networkmanager/interfaces.js

```diff
@@ -421,63 +443,52 @@
     /* NetworkManager specific data conversions and utility functions.
      */
 
-    function ip4_address_from_nm(addr) {
-        return [utils.ip4_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip4_to_text(addr[2], true)
-        ];
+    function ip_address_from_nm(addr) {
+        return {
+            address: addr.address.v,
+            prefix: utils.ip_prefix_to_text(addr.prefix.v)
+        };
     }
 
-    function ip4_address_to_nm(addr) {
-        return [utils.ip4_from_text(addr[0]),
-            utils.ip4_prefix_from_text(addr[1]),
-            utils.ip4_from_text(addr[2], true)
-        ];
-    }
+    function ip_address_to_nm(addr, ipv) {
+        const prefix = ipv === "ipv4" ? utils.ip4_prefix_from_text(addr.prefix) : utils.ip_prefix_from_text(addr.prefix);
 
-    function ip4_route_from_nm(addr) {
-        return [utils.ip4_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip4_to_text(addr[2], true),
-            utils.ip_metric_to_text(addr[3])
-        ];
-    }
+        if (!utils.validate_ip(addr.address)) {
+            throw cockpit.format(_("Invalid IP address: $0"), addr.address);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-1fe4fb60-20250903-092011-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)